eessi/testsuite/tests/apps/QuantumESPRESSO.py

```diff
@@ -0,0 +1,108 @@
+"""
+
+import reframe as rfm
+from hpctestlib.sciapps.qespresso.benchmarks import QEspressoPWCheck
+from reframe.core.builtins import (  # added only to make the linter happy
+    parameter, run_after)
+
+from eessi.testsuite import hooks
+from eessi.testsuite.constants import (COMPUTE_UNIT, CPU, DEVICE_TYPES, GPU,
+                                       SCALES, TAGS)
+from eessi.testsuite.utils import find_modules, log
+
+
+@rfm.simple_test
+class EESSI_QuantumESPRESSO_PW(QEspressoPWCheck):
+    scale = parameter(SCALES.keys())
+    valid_prog_environs = ['default']
+    valid_systems = ['*']
+    time_limit = '30m'
+    module_name = parameter(find_modules('QuantumESPRESSO'))
+    # For now, QE is being build for CPU targets only
+    # compute_device = parameter([DEVICE_TYPES[CPU], DEVICE_TYPES[GPU]])
+    compute_device = parameter([DEVICE_TYPES[CPU], ])
+
+    @run_after('init')
+    def run_after_init(self):
+        """Hooks to run after the init phase"""
+
+        # Filter on which scales are supported by the partitions defined in the ReFrame configuration
+        hooks.filter_supported_scales(self)
+
+        # Make sure that GPU tests run in partitions that support running on a GPU,
+        # and that CPU-only tests run in partitions that support running CPU-only.
+        # Also support setting valid_systems on the cmd line.
+        hooks.filter_valid_systems_by_device_type(self, required_device_type=self.compute_device)
+
+        # Support selecting modules on the cmd line.
+        hooks.set_modules(self)
+
+        # Support selecting scales on the cmd line via tags.
+        hooks.set_tag_scale(self)
+
+    @run_after('init')
+    def set_tag_ci(self):
+        """Set tag CI on smallest benchmark, so it can be selected on the cmd line via --tag CI"""
+        min_ecut = min(QEspressoPWCheck.ecut.values)
+        min_nbnd = min(QEspressoPWCheck.nbnd.values)
+        if self.ecut == min_ecut and self.nbnd == min_nbnd:
+            self.tags.add(TAGS['CI'])
+            log(f'tags set to {self.tags}')
+
+    @run_after('init')
+    def set_increased_walltime(self):
+        """Increase the amount of time for the largest benchmark, so it can complete successfully."""
+        max_ecut = max(QEspressoPWCheck.ecut.values)
+        max_nbnd = max(QEspressoPWCheck.nbnd.values)
+        if self.ecut == max_ecut and self.nbnd == max_nbnd:
+            self.time_limit = '60m'
+
+    @run_after('setup')
+    def run_after_setup(self):
+        """Hooks to run after the setup phase"""
+
+        # Calculate default requested resources based on the scale:
+        # 1 task per CPU for CPU-only tests, 1 task per GPU for GPU tests.
+        # Also support setting the resources on the cmd line.
+        if self.compute_device == DEVICE_TYPES[GPU]:
+            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT[GPU])
+        else:
+            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT[CPU])
```

Comment:
```suggestion
            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT[CPU])

    @run_after('setup')
    def set_mem(self):
        """Setting memory requirements for the test (8GB minumum to 1GB per core)"""
        mem_gb = max(8, self.num_tasks)
        self.extra_resources = {'memory': {'size': '%sGB' % mem_gb}}

```

@casparvl Do you think this would work or will it affect the sites other than VEGA?

Reply:
It could, potentially. 

Suppose you have a system which has nodes with 256 cores and 128GB memory. In principle, that's sufficient memory to run with 256 tasks per node (and 1 core per task) - I found that the `MaxRSS` for a task count of 256 was around 64 GB. However, your code above would request a full 256 GB. SLURM would then reject the job saying that no nodes are available that satisfy the request. 

Let's see what @smoors thinks about my pseudo-code in https://github.com/EESSI/test-suite/pull/128#issuecomment-2107992627 . If he thinks that's a reasonable approach, the only thing left to do then is come up with a reasonable formula for the memory requirement as function of the task count...

One more thing we should remember: we should check if the `'available_memory_per_node'` field is defined in the ReFrame config and if not, print a sensible error. Something similar to https://github.com/EESSI/test-suite/blob/a1021d793bfa45fcf889aaf87f2cb1f0ec324282/eessi/testsuite/utils.py#L115 but then for `extras` dictionary entries.